player/main.c

```diff
@@ -171,6 +171,10 @@ void mp_destroy(struct MPContext *mpctx)
     mp_uninit_ipc(mpctx->ipc_ctx);
     mpctx->ipc_ctx = NULL;
 
+#if HAVE_COCOA
+    cocoa_cb_animation_wait();
+#endif
+
```

Comment:
Can't this be somewhere out of the common code?

Reply:
i would like that too, but we have to block the core somehow till the animation is finished. think there is a good place to put it instead?

Reply:
Why does the core need to be blocked for that?

Reply:
if mpv closes mid animation cocoa can't recover the proper/former state anymore. this leads, in the case of the fs animation, to a always hidden Dock for example.

Reply:
Can't this be hooked to vo_opengl-cb uninit instead?

Reply:
Which mutex/cond?

Reply:
sry, i meant replacing [this](https://github.com/Akemi/mpv/blob/macOS_cocoa-cb2/video/out/cocoa-cb/window.swift#L182) NSCondition with a mutex/cond and moving it to vo_opengl-cb.

Reply:
this is a bit tricky. the events that are triggered at the end of the animation come from the main thread. so waiting in the vo_opengl-cb uninit function deadlocks everything. i could though, start the unlock on a different thread with a delay before the lock, since i know the theoretical duration of the animation. though this a bit hacky and i kinda dislike it.

Reply:
like suggested on irc i could do this differently and remove the waiting from the common code.